test/e2e/data/infrastructure-docker/main/cluster-template-dualstack/kcp-dualstack.yaml

```diff
@@ -0,0 +1,21 @@
+---
+kind: KubeadmControlPlane
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+metadata:
+  name: "${CLUSTER_NAME}-control-plane"
+spec:
+  kubeadmConfigSpec:
+    clusterConfiguration:
+      apiServer:
+        # host.docker.internal is required by kubetest when running on MacOS because of the way ports are proxied.
+        certSANs: [localhost,"127.0.0.1", "0.0.0.0", "::", "::1", host.docker.internal]
+    initConfiguration:
+      localAPIEndpoint:
+        advertiseAddress: '::' # Note: The control plane node should use an ipv6 address as an internalIP. The worker node should use an ipv4 address as an internalIP.
```

Comment:
Do we have more information on why control plane / worker nodes should use IPv6 vs IPv4?

Reply:
I assume it's because of this? https://github.com/kubernetes-sigs/cluster-api/pull/8517/files#r1164564959

Did I get it  correctly that by setting this here to :: and 0.0.0.0 for the worker we say that they should use either IPv6 or IPv4 and they Nodes can't use both? (or is it only that kubeadm only takes one, but the Node still ends up with an IPv4 and an IPv6 adress?)

Reply:
> Do we have more information on why control plane / worker nodes should use IPv6 vs IPv4?
(To close this loop)  This is just to ensure we have mixed IPs in this Cluster. Working on an update to make CAPD support true dual-stack nodes.